Analysis/Tasks/PWGHF/taskD0D0barCorrelation.cxx

```diff
@@ -0,0 +1,823 @@
+// or submit itself to any jurisdiction.
+
+/// \file taskD0D0barCorrelation.cxx
+/// \brief D0-D0bar analysis task - data-like, MC-reco and MC-kine analyses. For ULS and LS pairs
+///
+/// \author Fabio Colamaria <fabio.colamaria@ba.infn.it>, INFN Bari
+
+#include <cmath>
+
+#include "Framework/AnalysisTask.h"
+#include "Framework/HistogramRegistry.h"
+#include "AnalysisDataModel/HFSecondaryVertex.h"
+#include "AnalysisDataModel/HFCandidateSelectionTables.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::aod::hf_cand_prong2;
+using namespace o2::framework::expressions;
+using namespace o2::constants::math;
+
+// the following is needed to extend the standard candidate table, and allow grouping candidate table by collisions
+namespace o2::aod
+{
+namespace hf_2prong_correlation
+{
+DECLARE_SOA_INDEX_COLUMN(Collision, collision);
+} // namespace hf_2prong_correlation
+DECLARE_SOA_TABLE(HF2ProngCollis, "AOD", "COLLID_2PR", aod::hf_2prong_correlation::CollisionId);
+
+using Big2Prong = soa::Join<aod::HfCandProng2, aod::HFSelD0Candidate, aod::HF2ProngCollis>;
+using Big2ProngMC = soa::Join<aod::HfCandProng2, aod::HFSelD0Candidate, aod::HfCandProng2MCRec, aod::HF2ProngCollis>;
+} // namespace o2::aod
+
+// preliminary task to fill the column index to the extended candidate table
+struct CreateBig2Prong {
+
+  Produces<aod::HF2ProngCollis> create2ProngIndexCollColumn;
+  void process(aod::HfCandProng2 const& candidates, aod::Tracks const& tracks)
+  {
+    for (auto& candidate : candidates) {
+      int indexColl = candidate.index0_as<aod::Tracks>().collisionId(); //takes index of collision from first D daughter
+      create2ProngIndexCollColumn(indexColl);
+    }
+  }
+};
+
+void customize(std::vector<o2::framework::ConfigParamSpec>& workflowOptions)
+{
+  ConfigParamSpec optionDoMC{"doMC", VariantType::Bool, false, {"Run MC-dedicated tasks."}};
+  workflowOptions.push_back(optionDoMC);
+}
+
+#include "Framework/runDataProcessing.h"
+
+double getDeltaPhi(double phiD, double phiDbar)
+{
+  double dPhi = phiDbar - phiD;
+
+  if (dPhi < -o2::constants::math::PI / 2.)
+    dPhi = dPhi + 2. * o2::constants::math::PI;
+  if (dPhi > 3. * o2::constants::math::PI / 2.)
+    dPhi = dPhi - 2. * o2::constants::math::PI;
+
+  return dPhi;
+}
+
+double evaluatePhiByVertex(double xVertex1, double xVertex2, double yVertex1, double yVertex2)
+{
+  double phi = std::atan2((yVertex2 - yVertex1), (xVertex2 - xVertex1));
+
+  if (phi < 0.)
+    phi = phi + 2. * o2::constants::math::PI;
+  if (phi > 2. * o2::constants::math::PI)
+    phi = phi - 2. * o2::constants::math::PI;
+
+  return phi;
+}
+
+/// D0 analysis task - for real data and data-like analysis (i.e. reco-level w/o matching request via MC truth)
+struct TaskD0D0barCorrelation {
+
+  double maxEtaCut = 5., PtThresholdForMaxEtaCut = 10.; //for hDDbarVsEtaCut, gives eta increment of 0.1 and pt thr increments of 0.5
```

Comment:
1) Yes because cutEtaCandMax is a selection done on the candidates (which at some points could be changed to a y selection), while maxEtaCut defines the limit of the scan for the potential detector acceptance
2) so I was initially in doubt, but I think that with values large enough, as the current, they can be kept as constant (modified it, but moved outside the tasks to avoid multiple definition)
3) yes right, it's a leftover from a previous version, with dynamic increment depending on those variables, I forgot changing the comment
4) ok, modified